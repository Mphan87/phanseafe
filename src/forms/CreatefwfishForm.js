import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import "./form.css";


function CreatefwfishForm({createfwfish}) {
  const navigate = useNavigate();
  const [formData, setFormData] = useState({
    handle: "",
    name: "",
    taxonomy: "",
    image: "",
    description: "",
    maxsize: "",
    type: "",
  });


  const [formErrors, setFormErrors] = useState([]);
  async function handleSubmit(evt) {
    evt.preventDefault();
    let result = await createfwfish(formData);
    console.log(result)
    if (result.success) {
      console.log(result.success)
      navigate("/freshwaterfish");
    } else {
      setFormErrors(result.errors);
    }
  }

  function handleChange(evt) {
    const { name, value } = evt.target;
    setFormData(data => ({ ...data, [name]: value }));
    console.log(setFormData)
  }
  
  return (
    <div>
    <h3 className="fs-title">Sign Up</h3>
    <form id="msform" onSubmit={handleSubmit}>

        <div className="form-group">
          <label>Handle</label>
          <input
            name="handle"
            className="form-control"
            value={formData.handle.replace(/ /g,"-")}
            onChange={handleChange}
          />
        </div>

        <div className="form-group">
          <label>Name</label>
          <input
            name="name"
            className="form-control"
            value={formData.name}
            onChange={handleChange}
          />
        </div>

        <div className="form-group">
          <label>Taxonomy</label>
          <input
            name="taxonomy"
            className="form-control"
            value={formData.taxonomy}
            onChange={handleChange}
          />
        </div>
        <div className="form-group">
          <label>Image</label>
          <input
            name="image"
            className="form-control"
            value={formData.image}
            onChange={handleChange}
          />
        </div>
        <div className="form-group">
          <label>Description</label>
          <input
            name="description"
            className="form-control"
            value={formData.description}
            onChange={handleChange}
          />
        </div>
        <div className="form-group">
          <label>Maxsize</label>
          <input
            name="maxsize"
            className="form-control"
            value={formData.maxsize}
            onChange={handleChange}
          />
        </div>



        <div className="form-group">
          <form id="dropdown">
          <label>Type:</label>
            <select name="type" onChange={handleChange}>
              <option value="Lake-Malawi">Lake Malawi</option>
              <option value="South-American-cichlids">South American cichlids</option>
              <option value="Tetras">Tetras</option>
              <option value="Loaches">Loaches</option>
              <option value="Live-bearers">Live bearers</option>
              <option value="Central-American-cichlids">Central American cichlids</option>
            </select>
          </form>
        </div>


        <button
          type="submit"
          className="next action-button"
          onSubmit={handleSubmit}
        >
          Submit
        </button>
      </form>
    </div>
  );
}

export default CreatefwfishForm;