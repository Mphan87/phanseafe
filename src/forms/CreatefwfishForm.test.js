import React from "react";
import { render } from "@testing-library/react";
import CreatefwfishForm from "./CreateswfishForm";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <CreatefwfishForm />
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
