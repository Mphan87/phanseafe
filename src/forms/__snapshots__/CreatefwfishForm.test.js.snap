// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3
      class="fs-title"
    >
      Sign Up
    </h3>
    <form
      id="msform"
    >
      <div
        class="form-group"
      >
        <label>
          Handle:
        </label>
        <input
          class="form-control"
          name="handle"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Name:
        </label>
        <input
          class="form-control"
          name="name"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Taxonomy:
        </label>
        <input
          class="form-control"
          name="taxonomy"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Image:
        </label>
        <input
          class="form-control"
          name="image"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Reefsafe:
        </label>
        <input
          class="form-control"
          name="reefsafe"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Description:
        </label>
        <input
          class="form-control"
          name="description"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Maxsize:
        </label>
        <input
          class="form-control"
          name="maxsize"
          value=""
        />
      </div>
      <div
        class="form-group"
      >
        <label>
          Type:
        </label>
        <select
          name="type"
        >
          <option
            value="Clownfish"
          >
            Clownfish
          </option>
          <option
            value="Tangs"
          >
            Tangs
          </option>
          <option
            value="Damselfish"
          >
            Damselfish
          </option>
          <option
            value="Angelfish_large"
          >
            Large Angel Fish
          </option>
          <option
            value="Angelfish_dwarf"
          >
            Angelfish dwarf
          </option>
          <option
            value="Blennies"
          >
            Blennies
          </option>
        </select>
      </div>
    </form>
    <button
      class="next action-button"
      type="submit"
    >
      Submit
    </button>
  </div>
</DocumentFragment>
`;
